swap_meet/vendor.py

```diff
@@ -1,2 +1,86 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self,inventory=None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self,item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self,item):
+        try:
+            self.inventory.remove(item)
+            return item
+        except ValueError:
+            return False
+
+    def get_by_category(self,category):
+        return [item for item in self.inventory if item.category == category]
+
+    def get_by_age(self):
+        min_age = float("inf")
+        min_items = [None]
+        for item in self.inventory:
+            if item.age < min_age:
+                min_items[0], min_age = item, item.age
+            if item.age == min_age:
+                min_items.append(item)
+        return min_items
+    
+    def swap_items(self,other,my_item,their_item):
+        if my_item not in self.inventory or their_item not in other.inventory:
+            return False
+        self.remove(my_item)
+        self.add(their_item)
+        other.remove(their_item)
+        other.append(my_item)
+        return True
+
+    def swap_first_item(self,other):
+        if self.inventory == [] or other.inventory == []:
+            return False
+        my_item = self.inventory[0]
+        their_item = other.inventory[0]
+        # self.swap_items(other,my_item,their_item)
+        self.inventory.append(their_item)
+        other.inventory.append(my_item)
+        del self.inventory[0] # O(1) vs O(n) in swap_items as we know the index
+        del other.inventory[0] # O(1) vs O(n) in swap_items as we know the index
```

Comment:
Deleting an item from a list is still an O(n) operation, because when an item is deleted from a list all of the subsequent items must be shifted up one position.  With that in mind, using swap_items would be the same time complexity and also DRY the code here.